### PR TITLE
Made configuration of NOO and FLP resources optional

### PR DESCRIPTION
**Pull request by nathan-weinberg**
Tested here: https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/scale-ci/job/nweinber-e2e-benchmarking-multibranch-pipeline/job/netobserv-perf-tests/78/

### CODE REVIEW

**Comment by nathan-weinberg** [comment]
Additional test here: https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/scale-ci/job/nweinber-e2e-benchmarking-multibranch-pipeline/job/netobserv-perf-tests/83/

File: Jenkinsfile [hunk 1]
```diff
@@ -146,17 +146,17 @@ pipeline {
         )
         string(
             name: 'FLP_SAMPLING_RATE',
-            defaultValue: '1',
+            defaultValue: '',
             description: 'Rate at which to sample flows'
         )
```
Comment (memodi) [inline Jenkinsfile:151]: I think defaulting sampling to 1 is fine, we should mostly use 1 for our testing.

File: Jenkinsfile [hunk 2]
```diff
@@ -460,24 +460,46 @@ pipeline {
...
+                        """)
+                        if (returnCode.toInteger() != 0) {
+                            error('Updating controller memory limit failed :(')
+                        }
                     }
-                    else {
-                        println 'Successfully updated common parameters of NOO and flowcollector :)'
+                    if (params.FLP_SAMPLING_RATE != '') {
+                        returnCode = sh(returnStatus: true, script: """
+                            oc patch flowcollector cluster --type=json -p "[{"op": "replace", "path": "/spec/agent/ebpf/sampling", "value": ${params.FLP_SAMPLING_RATE}}] -n netobserv"
```
Comment (memodi) [inline Jenkinsfile:478]: always patch for sampling to bring it to 1.

**Review by memodi — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (nathan-weinberg) [reply Jenkinsfile:151]: Ack, will leave that default in

File: Jenkinsfile (hunk 2 above)
Comment (memodi) [reply Jenkinsfile:478]: in that case you'd always have to patch, no? so no need for if condition there since `FLP_SAMPLING_RATE` will always be set..

**Pull request merged**